versioned/persist/tests/src/main/java/org/projectnessie/versioned/persist/tests/AbstractDatabaseAdapterTest.java

```diff
@@ -33,6 +33,10 @@
 public abstract class AbstractDatabaseAdapterTest {
   @NessieDbAdapter protected static DatabaseAdapter databaseAdapter;
 
+  protected boolean commitWritesValidated() {
```

Comment:
personally not a fan of the iceberg style naming...(i.e. a method starting with "commit" not being an action)
how about `hasExistenceCheckOnCommitWrite` ?

and maybe add short javadoc what this means for a DB adapter...

but this makes me wonder, should this method maybe exist on DB-adapters directly? :thinking:
in a way it describes whether the commit-write methods of a DB-adapter really can throw the `ReferenceNotFoundException` as claimed by the method signature.
WDYT?

Reply:
As for me the name sounds fine ("commit" can be a noun too).

Moving the method to the `DatabaseAdapter` interface sounds like a good idea to me because it's the adapter implementation that controls this behaviour. Are there any adverse effects of that?

Reply:
Users of DA should _never_ assume anything derived from that fact

Reply:
If not `DatabaseAdapter`, maybe `DatabaseAdapterFactory`?

Reply:
No - whether the database is transactional or cannot "insert" the same key again is an internal concern

Reply:
fair enough.

Reply:
to some degree that internal concern is forced onto the user of the adapter by having to handle `ReferenceNotFoundException` even though there are no good rules on when it happens, what exactly the problem was nor how to appropriately handle it.
i guess its just not easy to model this nicely with the many db adapters that we have.